core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -446,46 +448,127 @@ object StateChangeHandlers {
 }
 
 /**
-  * Acls for resources are stored in ZK under a root node that is determined by the [[ResourceNameType]].
-  * Under each [[ResourceNameType]] node there will be one child node per resource type (Topic, Cluster, Group, etc).
-  * Under each resourceType there will be a unique child for each resource path and the data for that child will contain
+  * Acls for resources are stored in ZK under two root paths:
+  * <ul>
+  *   <li>[[org.apache.kafka.common.resource.ResourceNameType#LITERAL Literal]] patterns are stored under '/kafka-acl'.
+  *   The format is JSON. See [[kafka.zk.ResourceZNode]] for details.</li>
+  *   <li>All other patterns are stored under '/kafka-acl-extended/<i>pattern-type</i>'.
+  *   The format is JSON. See [[kafka.zk.ResourceZNode]] for details.</li>
```

Comment:
We could store literal patterns under the new path, right?

It's more like, the old path can only have LITERAL patterns, but the new path can have that and other stuff.

Reply:
If we stored any literal under the new path old brokers wouldn't pick them up. So for now all literal go under the old path.